ddl/ddl_api.go

```diff
@@ -907,29 +907,34 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 		return errors.Trace(err)
 	}
 
-	if pi != nil {
-		if err = checkPartitionNameUnique(tbInfo, pi); err != nil {
-			return errors.Trace(err)
-		}
+	if pi != nil && pi.Type == model.PartitionTypeRange {
+		// Check partition by range.
+		if s.Partition.ColumnNames == nil {
```

Comment:
From the title, I can tell there is only one constraint on the checking "restrict the check partition function to check only the rang partition". But I find another constraint here.